src/components/calcite-switch/calcite-switch.tsx

```diff
@@ -60,109 +141,85 @@ export class CalciteSwitch {
       (!this.el.closest("label") && e.target === this.el)
     ) {
-      this.updateSwitch(e);
+      this.toggle();
     }
   }
 
-  @Listen("keydown") keyDownHandler(e: KeyboardEvent): void {
+  @Listen("keydown")
+  keyDownHandler(e: KeyboardEvent): void {
     const key = getKey(e.key);
     if (!this.disabled && (key === " " || key === "Enter")) {
-      this.updateSwitch(e);
+      this.toggle();
     }
   }
 
-  @Watch("switched") switchWatcher(): void {
-    this.switched
-      ? this.inputProxy.setAttribute("checked", "")
-      : this.inputProxy.removeAttribute("checked");
-  }
-
-  private inputProxy: HTMLInputElement;
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
 
   connectedCallback(): void {
-    this.setupProxyInput();
+    this.guid = this.el.id || `calcite-switch-${guid()}`;
   }
 
-  disconnectedCallback(): void {
-    this.observer.disconnect();
+  componentDidLoad(): void {
+    this.renderInput();
   }
 
-  componentWillRender(): void {
-    this.syncProxyInputToThis();
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  private renderInput(): void {
+    this.switched && this.inputEl.setAttribute("checked", "");
+    this.inputEl.disabled = this.disabled;
+    this.inputEl.id = `${this.guid}-input`;
+    this.inputEl.name = this.name;
+    this.inputEl.style.setProperty("bottom", "0", "important");
+    this.inputEl.style.setProperty("left", "0", "important");
+    this.inputEl.style.setProperty("margin", "0", "important");
+    this.inputEl.style.setProperty("opacity", "0", "important");
+    this.inputEl.style.setProperty("outline", "none", "important");
+    this.inputEl.style.setProperty("padding", "0", "important");
+    this.inputEl.style.setProperty("position", "absolute", "important");
+    this.inputEl.style.setProperty("right", "0", "important");
+    this.inputEl.style.setProperty("top", "0", "important");
+    this.inputEl.style.setProperty("transform", "none", "important");
+    this.inputEl.style.setProperty("-webkit-appearance", "none", "important");
+    this.inputEl.style.setProperty("z-index", "-1", "important");
+    this.inputEl.type = "checkbox";
+    if (this.value) {
+      this.inputEl.value = this.value;
+    }
+    this.el.appendChild(this.inputEl);
   }
 
   render(): VNode {
     const dir = getElementDir(this.el);
-
     return (
-      <Host
-        aria-checked={this.switched.toString()}
-        dir={dir}
-        tabIndex={this.disabled ? -1 : this.tabIndex}
-      >
-        <div class="track">
-          <div class="handle" />
+      <Host>
+        <div
+          aria-checked={this.switched.toString()}
+          class="container"
+          dir={dir}
```

Comment:
Can we use a class instead of the dir attribute here? I don't think this CSS is working currently since the styles in the CSS are specified for host direction.

Reply:
```
import { CSS_UTILITY } from "../../utils/resources";

class={{ [CSS_UTILITY.rtl]: dir === "rtl" }}
```

```
.calcite--rtl calcite-switch {
  margin-left: 0;
  margin-right: var(--calcite-spacing-half);
}
```